### PR TITLE
SW-1755: Add Species Tooltips

### PR DESCRIPTION
**Pull request by joshbuchea**
This PR updates several components to support work for the [Web-App Tooltips Epic](https://terraformation.atlassian.net/browse/SW-1151).

The primary change is adding an optional icon tooltip in the label of several form field components:

- `Dropdown`
- `Select`
- `TextField`
- `TableHeaderItem`

Please see the storybook deployment to review changes 🙏

### CODE REVIEW

**Review by karthikbtf — APPROVED** [review]

**Comment by charlie-tf** [comment]
I have a minor request: I noticed that the `marginLeft: '3px'` [here](https://github.com/terraware/web-components/blob/main/src/components/IconTooltip/index.tsx#L12) was decentering the tooltip arrow. Would you mind removing as part of this PR?

File: src/components/Textfield/Textfield.tsx [hunk 1]
```diff
@@ -110,15 +111,7 @@ export default function TextField(props: Props): JSX.Element {
...
-        {tooltipText !== undefined && (
-          <div className='textfield-label--icon-tooltip-container'>
-            <Tooltip title={tooltipText} placement='top' arrow={true}>
-              <span>
-                <Icon name='info' className='textfield-label--icon-tooltip'/>
-              </span>
-            </Tooltip>
-          </div>
-        )}
+        {tooltipTitle && <IconTooltip title={tooltipTitle} />}
```
Comment (charlie-tf) [inline src/components/Textfield/Textfield.tsx:114]: this should have `placement={'top'}` to match previous implementation

**Comment by joshbuchea** [comment]
@charlie-tf the `{ marginLeft: '3px'; marginTop: '-1px' }` styles were necessary to match the border radius in the design specs. Adding the border radius created a visual gap between the arrow and the rest of the tool tip. The styles were added to remove the visual gap between the arrow & tooltip.

File: src/components/Textfield/Textfield.tsx (hunk 1 above)
Comment (joshbuchea) [reply src/components/Textfield/Textfield.tsx:114]: Nice catch, thanks! 🙌

**Comment by charlie-tf** [comment]
> @charlie-tf the `{ marginLeft: '3px'; marginTop: '-1px' }` styles were necessary to match the border radius in the design specs. Adding the border radius created a visual gap between the arrow and the rest of the tool tip. The styles were added to remove the visual gap between the arrow & tooltip.

@joshbuchea Oh, I think I understand what you mean. I'm not seeing it in the storybook, though. It looks like the arrow has plenty of distance from the radius when I took out the 3px here:
image

File: src/components/Textfield/Textfield.tsx (hunk 1 above)
Comment (charlie-tf) [reply src/components/Textfield/Textfield.tsx]: sigh, sorry for coming back to this again. Looks like we'll need to add a string? tooltip placement prop to TextField (defaulted to 'top'). The figma shows all the added tooltips as 'top-start', but we'll want to have existing tooltips 'top' I guess.

**Comment by joshbuchea** [comment]
@charlie-tf the gap didn't show for all variations of the `placement` value, but did show for some. I'm not in favor of adding random/magic numbers, but found it necessary to match designs in this case.

**Pull request merged**